docs/operations/hook-sidecar.md

```diff
@@ -0,0 +1,168 @@
+communication with the main container.
+
+The image contains the `sidecar-shim` binary built using
+[`sidecar_shim.go`](https://github.com/kubevirt/kubevirt/blob/main/cmd/sidecars/sidecar_shim.go) which should be kept
+as the entrypoint of the container. This binary will search in `$PATH` for binaries named after the hook names (e.g
+`onDefineDomain` and `preCloudInitIso`) and run them. User must provide the necessary arguments as command line options
+(flags).
+
+In the case of `onDefineDomain`, the arguments will be the VMI information as JSON string, (e.g `--vmi vmiJSON`) and
+the current domain XML (e.g `--domain domainXML`). As standard output it expects the modified domain XML.
+
+In the case of `preCloudInitIso`, the arguments will be the VMI information as JSON string, (e.g `--vmi vmiJSON`) and
+the CloudInitData (e.g `--cloud-init cloudInitJSON`). As standard output it expects the modified CloudInitData (as
+JSON).
+
+Besides a binary, one could also execute shell or python scripts by making them available at the expected location.
+
+## Go, Python, Shell - pick any one
+
+Although a binary doesn't strictly need to be generated from Go code, and a script doesn't strictly need to be one
+among Shell or Python, for the purpose of this guide, we will use those as examples.
+
+### Go binary
+
+Example Go code modifiying the [SMBIOS system
+information](https://libvirt.org/formatdomain.html#smbios-system-information) can be found in the [KubeVirt
+repo](https://github.com/kubevirt/kubevirt/tree/main/cmd/sidecars/smbios). Binary generated from this code, when
+available under `/usr/bin/ondefinedomain` in the sidecar-shim-image, is run right before VMI creation and the baseboard
+manufacturer value is modified to reflect what's provided in the `smbios.vm.kubevirt.io/baseBoardManufacturer`
+annotation in [VMI spec](https://github.com/kubevirt/kubevirt/blob/main/examples/vmi-with-sidecar-hook.yaml).
+
+### Shell or Python script
+
+If you pefer writing a shell or python script instead of a Go program, create a Kubernetes ConfigMap and use
+annotations to make sure the script is run before the VMI creation. The flow would be as below:
+
+1. Create a ConfigMap containing the shell or python script you want to run
+1. Create a VMI containing the annotation `hooks.kubevirt.io/hookSidecars` and mention the ConfigMap information in it.
+
+#### ConfigMap with shell script
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config-map
+data:
+  my_script.sh: |
+    #!/bin/sh
+    tempFile=`mktemp --dry-run`
+    echo $4 > $tempFile
+    sed -i "s|<baseBoard></baseBoard>|<baseBoard><entry name='manufacturer'>Radical Edward</entry></baseBoard>|" $tempFile
+    cat $tempFile
+```
+
+#### ConfigMap with python script
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config-map
+data:
+  my_script.sh: |
+    #!/usr/bin/env python
```

Comment:
The sidecar-shim image has `python3` but not `python`. @victortoso and I noticed this when debugging together. Just a reminder to update this.